packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -1083,6 +1058,13 @@ export class GarbageCollector implements IGarbageCollector {
 			fromId: fromNodePath,
 			autorecovery,
 		});
+
+		// This node is referenced - Clear its unreferenced state
+		if (this.unreferencedNodesState.delete(toNodePath)) {
+			// If it was deleted, then it moved from unreferenced to referenced.
```

Comment:
It's not clear from this comment why this is needed. Looks like its only to generate the updated stats counts?